### DISCUSSION TITLE
how to stop sending the request after checking for a some condition in AddRequestTransform and send a response (non proxied custom)

### DISCUSSION THREAD

**Discussion by dotsource** [dotnet/yarp#907] [Q&A] [answered]
how to stop sending the request after checking for a some condition in AddRequestTransform and send a response (non proxied custom)

**Comment by Tratcher** [comment, marked as answer]
Transforms can't stop the request from being sent, try using middleware for that instead:
https://github.com/microsoft/reverse-proxy/blob/main/docs/docfx/articles/middleware.md#send-an-immediate-response

Out of curiosity, what kind of condition are you checking for, and what's the alternate response you send?

**Reply by dotsource** [reply]
Thank you very much for the response. Much appreciated. 

I am checking if the query parameter has a valid value and if not I was thinking of sending a response without sending it to upstream server. You are right that I could use a middleware for that but I am having trouble getting a asp.net middleware and proxy play well.

**Reply by Tratcher** [reply]
That linked middleware doc is brand new, let me know if you have trouble following it.
